cmd/storage-class.go

```diff
@@ -0,0 +1,255 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+const (
+	// metadata entry for storage class
+	amzStorageClass = "x-amz-storage-class"
+	// Reduced redundancy storage class
+	reducedRedundancyStorageClass = "MINIO_STORAGE_CLASS_RRS"
+	// Standard storage class
+	standardStorageClass = "MINIO_STORAGE_CLASS_STANDARD"
+	// Supported storage class scheme is EC
+	supportedStorageClassScheme = "EC"
+	// Minimum parity disks
+	minimumParityDisks = 2
+	defaultRRSParity   = 2
+
+	// data blocks
+	dataBlocks   = "dataBlocks"
+	parityBlocks = "parityBlocks"
+)
+
+// Struct to hold storage class
+type storageClass struct {
+	Scheme string
+	Parity int
+}
+
+type storageClassConfig struct {
+	Standard string `json:"standard"`
+	RRS      string `json:"rrs"`
+}
+
+// Parses given storageClassEnv and returns a storageClass structure.
+// Supported Storage Class format is "Scheme:Number of parity disks".
+// Currently only supported scheme is "EC".
+func mustParseStorageClass(storageClassEnv string) (sc storageClass, err error) {
+	s := strings.Split(storageClassEnv, ":")
+
+	// only two elements allowed in the string - "scheme" and "number of parity disks"
+	if len(s) > 2 {
+		return storageClass{}, errors.New("Too many sections in " + storageClassEnv)
+	} else if len(s) < 2 {
+		return storageClass{}, errors.New("Too few sections in " + storageClassEnv)
+	}
+
+	// only allowed scheme is "EC"
+	if s[0] != supportedStorageClassScheme {
+		return storageClass{}, errors.New("Unsupported scheme " + s[0] + ". Supported scheme is EC")
+	}
+
+	// Number of parity disks should be integer
+	parityDisks, err := strconv.Atoi(s[1])
+	if err != nil {
+		return storageClass{}, err
+	}
+
+	scPtr := &storageClass{
```

Comment:
Wy take pointer and return *scPtr in line 83 -- might just as well use 'in value' (or better yet, store immediately in sc return parameter)

Reply:
agreed.